src/main/java/com/google/cloud/teleport/splunk/SplunkEventWriter.java

```diff
@@ -264,16 +266,35 @@ private void logWriteFailures(@StateId(COUNT_STATE_NAME) ValueState<Long> countS
     LOG.error("Failed to write {} events", countState.read());
     FAILED_WRITES.inc(countState.read());
   }
+  
+  /**
+   * Utility method to flush failed write events.
+   *
+   * @param builder {@link SplunkWriteError.Builder} object
+   * @param events List of {@link SplunkEvent}s to un-batch
+   * @param receiver Receiver to write {@link SplunkWriteError}s to
+   */
+  private void flushWriteFailures(
+      SplunkWriteError.Builder builder,
```

Comment:
We basically just need the statusCode and statusMessage. Lets see if we can find a way to just pass those (either individually or wrapped inside of a custom object). We shouldnt need to pass a builder. SplunkWriteError can be built inside this method.

Reply:
I'll change this to accept the statusCode and statusMessage along with the list of SplunkEvents and the output reciever. We'd then be able to build and output SplunkWriteErrors within the method.